src/arraystring.rs

```diff
@@ -658,49 +534,18 @@ where
     /// # }
     /// ```
     #[inline]
-    pub fn try_push(&mut self, character: char) -> Result<(), OutOfBounds> {
+    pub fn try_push(&mut self, ch: char) -> Result<(), OutOfBounds> {
         trace!("Push: {}", character);
-        let new_end = character.len_utf8().saturating_add(self.len().into());
-        is_inside_boundary(new_end, Self::capacity())?;
-        unsafe { self.push_unchecked(character) };
-        Ok(())
-    }
-
-    /// Inserts character to the end of the `ArrayString` assuming length is appropriate
-    ///
-    /// # Safety
-    ///
-    /// It's UB if `self.len() + character.len_utf8()` > [`capacity`]
-    ///
-    /// [`capacity`]: ./struct.ArrayString.html#method.capacity
-    ///
-    /// ```rust
-    /// # use arraystring::{Error, prelude::*};
-    /// # fn main() -> Result<(), Error> {
-    /// let mut s = ArrayString::<23>::try_from_str("My String")?;
-    /// unsafe { s.push_unchecked('!') };
-    /// assert_eq!(s.as_str(), "My String!");
-    ///
-    /// // s = ArrayString::<23>::try_from_str(&"0".repeat(ArrayString::<23>::capacity().into()))?;
-    /// // Undefined behavior, don't do it
-    /// // s.push_unchecked('!');
-    /// # Ok(())
-    /// # }
-    /// ```
-    #[inline]
-    pub unsafe fn push_unchecked(&mut self, ch: char) {
-        let (len, chlen) = (self.len(), ch.len_utf8().into_lossy());
-        debug!("Push unchecked (len: {}): {} (len: {})", len, ch, chlen);
-        encode_char_utf8_unchecked(self, ch, len);
-        self.size = self.size.saturating_add(chlen);
+        let mut buf = [0; 4];
+        self.try_push_str(ch.encode_utf8(&mut buf))
```

Comment:
Way better, no need to use unsafe for 4 bytes, thank you!

Reply:
yeah, took a little digging in the docs to find :)